Connection.php

```diff
@@ -647,8 +658,38 @@ public function executeCommand($name, $params = [])
         }
 
         \Yii::trace("Executing Redis Command: {$name}", __METHOD__);
-        fwrite($this->_socket, $command);
+        if ($this->retries > 0) {
+            $tries = $this->retries;
+            while ($tries-- >= 0) {
```

Comment:
This looks incorrect. If `$tries = 1` then `$tries--` will return `1`.  So `$this->retries = 1` will result 2 retries instead of 1.

It should be `while ($tries-- > 0)` or use pre-increment.

Reply:
if `$tries = 1`, the while loop will run once. that means on first exception, it will try to reconnect, then not enter the while loop again so the second command will run normally. This will be 2 "tries" but one of them is the **re**try ;)

Imo that is correct and expected.  `$this->retries = 0` means one command execution, `$this->retries = 1` means two in case of error.

Reply:
> if $tries = 1, the while loop will run once.

It will run twice: https://3v4l.org/NObVY

Or I'm missing something obvious. :P

Reply:
```
while (--$tries >= 0) {
    $i++;
}
```

Runs once 🤓 

https://3v4l.org/UhA6J

Reply:
Seems I mixed up the combination of `--` before or after and the `>` and `>=` usage.

- `--$tries` decrements first and then checks against 0
- `$tries--` checks against 0 and then decrements

Will add a test to verify the number of trials.

Reply:
I'd make that explicit by separating decrement and condition.

Reply:
fixed.